src/storage/src/hummock/compactor/mod.rs

```diff
@@ -655,7 +662,15 @@ impl Compactor {
         let mut last_table_stats = TableStats::default();
         let mut last_table_id = None;
         let mut compaction_statistics = CompactionStatistics::default();
+        let mut progress_key_num: u64 = 0;
+        const PROGRESS_KEY_INTERVAL: u64 = 100;
         while iter.is_valid() {
+            progress_key_num += 1;
+
+            if let Some(task_progress) = task_progress.as_ref() && progress_key_num % PROGRESS_KEY_INTERVAL == 0 {
```

Comment:
suggest:
```rust
if let Some(task_progress) = task_progress.as_ref() && progress_key_num >= PROGRESS_KEY_INTERVAL {
   task_progress.inc_progress_key(progress_key_num);
   progress_key_num = 0
}
```

and then we can call this in the end of this function `task_progress.inc_progress_key(progress_key_num);`

Reply:
Good catch ! fix it